csharp/src/Microsoft.ML.OnnxRuntime/Training/CheckpointState.shared.cs

```diff
@@ -19,22 +19,16 @@ internal IntPtr Handle
             }
         }
 
-        /// <summary>
-        /// Creates CheckpointState by loading state from path.
-        /// <param name="checkpointPath"> absolute path to checkpoint file.</param>
-        /// </summary>
-        public CheckpointState(string checkpointPath)
-            : base(IntPtr.Zero, true)
+        private CheckpointState(IntPtr checkpointHandle)
+            : base(checkpointHandle, true)
         {
-            if (NativeTrainingMethods.TrainingEnabled())
-            {
-                var envHandle = OrtEnv.Instance().Handle; // just so it is initialized
-                LoadCheckpoint(checkpointPath);
-            }
-            else
-            {
-                throw new InvalidOperationException("Training is disabled in the current build. Please build ONNXRuntime from source with the build flags enable_training_apis. \n");
-            }
+        }
+
+        public enum PropertyType
```

Comment:
Can this be `private enum PropertyType: long` ?

Reply:
Thanks that worked. But had to make it public again to support it being accessible from `NativeTrainingMethods`.
Let me know if I could use a way to hide it from other `CheckpointState` access points.